### PR TITLE
Move custom AppearDelay into its own interface

### PR DESCRIPTION
**Pull request by peppy**
Avoids the requirement of adding a `null` `AppearDelay` to every usage of `IHasTooltip`

### CODE REVIEW

File: osu.Framework/Graphics/Cursor/IHasAppearDelay.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu-framework/master/LICENCE
+
+namespace osu.Framework.Graphics.Cursor
+{
+    /// <summary>
+    /// A tooltip which provides a custom delay until it appears, override the <see cref="TooltipContainer"/>-wide default.
```
Comment (Aergwyn) [inline osu.Framework/Graphics/Cursor/IHasAppearDelay.cs]: is there a typo here? Shouldn't it be `overriding`?

File: osu.Framework/Graphics/Cursor/IHasAppearDelay.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    /// </summary>
+    public interface IHasAppearDelay : IDrawable
+    {
+        /// <summary>
+        /// The delay until the tooltip should be displayed. Null means the <see cref="TooltipContainer.AppearDelay"/> from the <see cref="TooltipContainer"/> containing the <see cref="Drawable"/> implementing this interface will be used.
```
Comment (Aergwyn) [inline osu.Framework/Graphics/Cursor/IHasAppearDelay.cs]: Maybe this doesn't need to mention the `null`-case anymore? The interface documentation itself mentions this already.

File: osu.Framework/Graphics/Cursor/IHasAppearDelay.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu-framework/master/LICENCE
+
+namespace osu.Framework.Graphics.Cursor
+{
+    /// <summary>
+    /// A tooltip which provides a custom delay until it appears, override the <see cref="TooltipContainer"/>-wide default.
+    /// </summary>
+    public interface IHasAppearDelay : IDrawable
```
Comment (smoogipoo) [inline osu.Framework/Graphics/Cursor/IHasAppearDelay.cs]: Should inherit from `IHasTooltip`.

Comment (Tom94) [reply osu.Framework/Graphics/Cursor/IHasAppearDelay.cs]: I disagree, unless the interface is renamed to `IHasTooltipWithAppearDelay`

Comment (smoogipoo) [reply osu.Framework/Graphics/Cursor/IHasAppearDelay.cs]: Given the xmldoc, and given the namespacing of the interface, I believe it should inherit.

Would also be fine with naming it `IHasTooltipWithAppearDelay`.

**Pull request merged**